### ISSUE TITLE
`newTimeLocked` redeem script uses wrong endianess for locktime

### ISSUE THREAD

**Issue by dooglus** [OutCast3k/coinbin#60]
I want a locktime of `1472817600` (`0x57c969c0`), so I enter date/time `09/02/2016 05:00` (I'm in PDST) but the redeems script it generates starts:

    0457c969c0

I would expect the locktime to be reversed:

    04c069c957

https://en.bitcoin.it/wiki/Script tells me:

> The stacks hold byte vectors. When used as numbers, byte vectors are interpreted as _little-endian_ variable-length integers

When I try to submit the signed transaction to Bitcoin Core it complains:

    error code: -26
    error message:
    64: non-mandatory-script-verify-flag (Negative locktime)

presumably because the `0xc0...` value is interpreted as a negative. The `0xc0` should be the least significant byte, but it's being used as the most significant.

See [this reddit post](https://www.reddit.com/r/Bitcoin/comments/51coa8/legitimately_2634_mbtc_for_the_first_person_who/) for an example of someone having trouble due to this bug.

**Comment by dooglus** [comment]
It looks like commit 3ba79f0e3d85a13031a211744f085ff753172b1a broke this.

`coinjs.numToByteArray()` already uses the correct endianess, so there's no need to `reverse()` its output.

**Comment by dooglus** [comment]
I don't know this code well enough to contribute a pull request, but something like the following might work:

    diff --git a/js/coin.js b/js/coin.js
    index e139fef..ab5aeb7 100644
    --- a/js/coin.js
    +++ b/js/coin.js
    @@ -148,7 +148,7 @@
     		}
     
     		var s = coinjs.script();
    -		s.writeBytes(coinjs.numToByteArray(checklocktimeverify).reverse());
    +		s.writeBytes(coinjs.numToByteArray(checklocktimeverify));
     		s.writeOp(177);//OP_CHECKLOCKTIMEVERIFY
     		s.writeOp(117);//OP_DROP
     		s.writeBytes(Crypto.util.hexToBytes(pubkey));
    @@ -687,7 +687,7 @@
     					// ^ <unlocktime> OP_CHECKLOCKTIMEVERIFY OP_DROP <pubkey> OP_CHECKSIG ^
     					r = {}
     					r.pubkey = Crypto.util.bytesToHex(s.chunks[3]);
    -					r.checklocktimeverify = coinjs.bytesToNum(s.chunks[0].slice().reverse());
    +					r.checklocktimeverify = coinjs.bytesToNum(s.chunks[0].slice());
     					r.address = coinjs.simpleHodlAddress(r.pubkey, r.checklocktimeverify).address;
     					r.type = "hodl__";
     				}

**Comment by dooglus** [comment]
@OutCast3k 

> Ive just fixed the code that caused this bug in commit 3ba79f0

That is a different bug. The bug you fixed only happens 1 out of 16 times, but your fix makes the result wrong 100% of the time because it reverses the locktime bytes twice. Please re-read my comments. People are likely losing money because of this bug.

**Comment by OutCast3k** [comment]
Hey dooglus,

Thanks for your help on this matter.

I made a commit 3c1e957 (this morning, after you posted) which fixes the bug brought up in commit 3ba79f0

The lines I modified where the same as your suggestion:

`-       s.writeBytes(coinjs.numToByteArray(checklocktimeverify).reverse());
+       s.writeBytes(coinjs.numToByteArray(checklocktimeverify));`

and

`-                   r.checklocktimeverify = coinjs.bytesToNum(s.chunks[0].slice().reverse());
+                   r.checklocktimeverify = coinjs.bytesToNum(s.chunks[0].slice());`

After this update, i'm generating the follow redeemscript; `0440f9c857b1752102b6231cc602740c29436eafbb6448880f4058cc3d2745c709deee313104678277ac`

Which bitcoin-cli can decode to

` $ btc decodescript "0440f9c857b1752102b6231cc602740c29436eafbb6448880f4058cc3d2745c709deee313104678277ac"
{
  "asm": "1472788800 OP_CHECKLOCKTIMEVERIFY OP_DROP 02b6231cc602740c29436eafbb6448880f4058cc3d2745c709deee313104678277 OP_CHECKSIG",
  "type": "nonstandard",
  "p2sh": "3B7DrDfvwqeJBXiukGRUW93VckkNb27wq3"
}`

Notice the timestamp is now 1472788800 which decodes to Fri, 02 Sep 2016 04:00:00 GMT

It all seems good to me. Have i missed something, or did you miss interpret what i meant by my previous comment?
